### ISSUE TITLE
Cannot call typevar-bound class method from self

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1205]
**Describe the bug**
When calling a class method annotated to take a type variable bound to the class as its first argument inside an instance method using `self`, Pyright reports that it's unable to bind the callee:

```
Could not bind method "bar" because "Type[Foo]" is not assignable to parameter "cls"
  "Foo" is incompatible with "type[TFoo]"
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That an error would not be generated on line 13.

**Screenshots or Code**
```python
# pyright: strict

from typing import TypeVar

TFoo = TypeVar('TFoo', bound='Foo')

class Foo:
    @classmethod
    def bar(cls: type[TFoo]) -> TFoo:
        ...

    def baz(self) -> None:
        self.bar()

```

**VS Code extension or command-line**
Either, 1.1.88

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.89, which I just published. It will also appear in the next Pylance release.
